src/promptflow/tests/executor/unittests/_utils/test_dataclass_serializer.py

```diff
@@ -0,0 +1,209 @@
+import pytest
+from datetime import datetime
+from dataclasses import dataclass
+from typing import Dict, List
+from promptflow._core.generator_proxy import GeneratorProxy
+from promptflow._utils.dataclass_serializer import \
+    get_type, serialize, deserialize_dataclass, deserialize_value, assertEqual
+from promptflow.contracts.run_info import RunInfo, Status
+from promptflow._core.connection_manager import ConnectionManager
+from promptflow.storage.run_records import NodeRunRecord
+from unittest.mock import patch, Mock
+import sys
+
+
+def get_connection_dict():
+    return {
+        "azure_open_ai_connection": {
+            "type": "AzureOpenAIConnection",
+            "value": {
+                "api_key": "<azure-openai-key>",
+                "api_base": "<aoai-api-endpoint>",
+                "api_type": "azure",
+                "api_version": "2023-07-01-preview",
+            },
+        },
+        "custom_connection": {
+            "type": "CustomConnection",
+            "value": {
+                "api_key": "<your-key>",
+                "url": "<connection-endpoint>",
+            },
+            "module": "promptflow.connections",
+            "secret_keys": ["api_key"],
+        },
+    }
+
+
+@pytest.mark.unittest
+@pytest.mark.parametrize(
+    "type_input, expected",
+    [
+        (NodeRunRecord, NodeRunRecord),
+        ([NodeRunRecord], List[NodeRunRecord]),
+        (dict(a=NodeRunRecord), Dict[str, NodeRunRecord]),
+        (int, int),
+        (str, str),
+    ]
+)
+def test_get_type(type_input, expected):
+    assert get_type(type_input) == expected
+
+
+@pytest.mark.unittest
+def test_serialize_dataclass():
+    start_time = datetime(2023, 9, 4)
+    end_time = datetime(2023, 9, 4)
+    node_run_info = RunInfo(
+        node=None,
+        run_id=None,
+        flow_run_id=None,
+        status=Status.Completed,
+        inputs=None,
+        output=None,
+        metrics=None,
+        error=None,
+        parent_run_id=None,
+        start_time=start_time,
+        end_time=end_time,
+        index=0,
+    )
+    node_record = NodeRunRecord.from_run_info(node_run_info)
+    assert deserialize_value(serialize(node_run_info), RunInfo) == node_run_info
+    assert serialize(node_record) == node_record.serialize()
+
+
+@pytest.mark.unittest
+@pytest.mark.parametrize(
+    "value, value_type",
+    [
+        (datetime(2023, 9, 4), datetime),
+        (Status.Completed, Status),
+        ([1, 2, 3], List[int]),
+        ({"a": 1, "b": 2}, Dict[str, int]),
+        (1, int),
+        ("a", str),
+    ]
+)
+def test_serialize_value(value, value_type):
+    assert deserialize_value(serialize(value), value_type) == value
```

Comment:
I guess we may also assert the serialized value to make sure all kinds of types are serialized as expected.
We add these tests because we want to make sure the behavior is clear and expected, not only go through the code coverage.

Reply:
Added assertion for test_serialize_dataclass and test_serialize_value.